client/app/cloud/project/compute/infrastructure/flavor.service.js

```diff
@@ -0,0 +1,177 @@
+(() => {
+
+    const oldFlavorRegex = /eg|sp|hg|vps-ssd/;
+
+    class CloudFlavorService {
+
+        constructor ($filter, CLOUD_FLAVORTYPE_CATEGORIES, CLOUD_INSTANCE_CPU_FREQUENCY) {
+            this.$filter = $filter;
+            this.CLOUD_FLAVORTYPE_CATEGORIES = CLOUD_FLAVORTYPE_CATEGORIES;
+            this.CLOUD_INSTANCE_CPU_FREQUENCY = CLOUD_INSTANCE_CPU_FREQUENCY;
+        }
+
+        static isOldFlavor (flavorName) {
+            return oldFlavorRegex.test(flavorName);
+        }
+
+        static getFlavorTypes (flavors) {
+            return _.uniq(_.pluck(flavors, "type"));
```

Comment:
_.pluck is deprecated in Lodash4. You can, instead, use:
.uniq(.map(images, "type")

Reply:
done